### DISCUSSION TITLE
Pointer Events Issue, possible bug ?

### DISCUSSION THREAD

**Discussion by LaurentInSeattle** [AvaloniaUI/Avalonia#15802] [Q&A]
Hi all, 

I am listening to pointer events on a Rectangle shape using the following code: 

`
        this.eventingRectangle.PointerPressed += this.OnPointerPressed;
        this.eventingRectangle.PointerReleased += this.OnPointerReleased;
        this.eventingRectangle.PointerEntered += this.OnPointerEnter;
        this.eventingRectangle.PointerExited += this.OnPointerLeave; 
`
For now, the corresponding code is just a basic Debug.WriteLine and I can see the following, when I move the mouse around: 
- Entering the area => Entered, good
- LB Click Down => Pressed, good
- Leaving the area => Nothing (bad!) 
- Seconds or minutes later: LB Click Up => Released , then Exited 

WPF was giving me the proper sequence: Enter, Down, Leave, Up.
 
Any thing I could have done wrong ??? 
Any workaround ? 
This is on Win 10 / Visual Studio 2022 / .Net 8 with Avalonia 11.0.10 - Not checked on any other platform.

Thanks!

**Comment by stevemonaco** [comment]
Related: https://github.com/AvaloniaUI/Avalonia/discussions/14502#discussioncomment-8377590 

AFAIK, this is just how it works in Avalonia because of how pointer capture works. Not sure if this is due to some temporary limitation or if this is expected forever, so the team would need to comment. Reminder that Avalonia APIs cannot be 1:1 with WPF everywhere because of varying levels of platform capabilities.

**Reply by LaurentInSeattle** [reply]
Well, well, expecting events to be reported in order does not seem to me so much to ask... 
This is the behaviour we had since the heydays of Dinosaur MFC and Old Fart WinForms...

Turns out that I am using the same workaround than the one you suggested, checking if we are still "inside" on Pointer Up.
But... Still not good.

**Reply by IanRawley** [reply]
It's matching UWP to the best of my knowledge rather than WPF.  See:
https://github.com/AvaloniaUI/Avalonia/pull/15357

Well, intended to be.  The PR hasn't merged yet.
